api/src/test/java/org/pmiops/workbench/elasticsearch/ElasticFiltersTest.java

```diff
@@ -75,147 +85,123 @@ public void setUp() {
-            .selectable(true)
-            .parentId(7)
-            .path("6.7"));
+            .parentId(question.getId());
+    saveCriteriaWithPath(question.getPath(), answer);
 
     // drug tree
-    criteriaDao.save(
+    // Use jdbcTemplate to create/insert data into the ancestor table
+    // The codebase currently doesn't have a need to implement a DAO for this table
+    jdbcTemplate.execute(
+        "create table cb_criteria_ancestor(ancestor_id integer, descendant_id integer)");
+    jdbcTemplate.execute(
+        "insert into cb_criteria_ancestor(ancestor_id, descendant_id) values (19069022, 21600009)");
+    CBCriteria drugParent =
+        drugCriteria().code("A").conceptId("21600001").group(true).selectable(false).parentId(0);
+    saveCriteriaWithPath("", drugParent);
+    CBCriteria drug1 =
         drugCriteria()
-            .id(10)
-            .code("A")
-            .conceptId("21600001")
-            .group(true)
-            .selectable(false)
-            .parentId(0)
-            .path(""));
-    criteriaDao.save(
-        drugCriteria()
-            .id(11)
             .code("A01")
             .conceptId("21600002")
             .group(true)
             .selectable(true)
-            .parentId(10)
-            .path("10"));
-    criteriaDao.save(
+            .parentId(drugParent.getId());
+    saveCriteriaWithPath(drugParent.getPath(), drug1);
+    CBCriteria drug2 =
         drugCriteria()
-            .id(12)
             .code("A01A")
             .conceptId("21600003")
             .group(true)
             .selectable(true)
-            .parentId(11)
-            .path("10.11"));
-    criteriaDao.save(
+            .parentId(drug1.getId());
+    saveCriteriaWithPath(drug1.getPath(), drug2);
+    CBCriteria drug3 =
         drugCriteria()
-            .id(13)
             .code("A01AA")
             .conceptId("21600004")
             .group(true)
             .selectable(true)
-            .parentId(12)
-            .path("10.11.12"));
-    criteriaDao.save(
+            .parentId(drug2.getId());
+    saveCriteriaWithPath(drug2.getPath(), drug3);
+    CBCriteria drug4 =
         drugCriteria()
-            .id(14)
             .code("9873")
             .conceptId("19069022")
             .group(false)
             .selectable(true)
-            .parentId(13)
-            .path("10.11.12.13"));
+            .type(CriteriaType.RXNORM.toString())
+            .parentId(drug3.getId());
+    saveCriteriaWithPath(drug3.getPath(), drug4);
 
     leafParam2 =
         new SearchParameter()
             .conceptId(772L)
-            .type(TreeType.ICD9.toString())
-            .subtype(TreeSubType.CM.toString())
+            .domain(DomainType.CONDITION.toString())
+            .type(CriteriaType.ICD9CM.toString())
+            .ancestorData(false)
+            .standard(false)
             .group(false);
   }
 
+  @After
+  public void tearDown() {
+    jdbcTemplate.execute("drop table cb_criteria_ancestor");
```

Comment:
This test has `@DirtiesContext` - is this statement necessary?

Reply:
jdbcTemplate is used to create/insert data into the cb_criteria_ancestor table since the codebase currently doesn't have a need to implement a DAO for this table. Therefore the @DirtiesContext annotation seems to only recognized Hibernate persisted entities and their related tables. So the following tear down method needs to drop to table to keep from colliding with other test classes that also use this approach to create the cb_criteria_ancestor table. Not implementing this drop causes subsequent test suite runs to fail with a BadSqlGrammarException: StatementCallback; bad SQL grammar [create table cb_criteria_ancestor(ancestor_id integer, descendant_id integer)]; nested exception is org.h2.jdbc.JdbcSQLException: Table "CB_CRITERIA_ANCESTOR" already exists. Another approach could be to try catch the create of cb_criteria_ancestor and do nothing on BadSqlGrammarException. Added this comment to the tear down method.